mergify_engine/web/api/queues.py

```diff
@@ -317,5 +325,367 @@ async def repository_queues(
+        ..., description="The owner of the repository"
+    ),
+    repository: github_types.GitHubRepositoryName = fastapi.Path(  # noqa: B008
+        ..., description="The name of the repository"
+    ),
+    queue_name: rules.QueueName = fastapi.Path(  # noqa: B008
+        ..., description="The name of the queue"
+    ),
+    redis_cache: utils.RedisCache = fastapi.Depends(  # noqa: B008
+        redis.get_redis_cache
+    ),
+    redis_queue: utils.RedisQueue = fastapi.Depends(  # noqa: B008
+        redis.get_redis_queue
+    ),
+    installation_json: github_types.GitHubInstallation = fastapi.Depends(  # noqa: B008
+        security.get_installation
+    ),
+) -> responses.JSONResponse:
+
+    try:
+        raw_json = await request.json()
+    except json.JSONDecodeError:
+        return responses.JSONResponse(
+            status_code=400, content={"error": "Invalid json"}
+        )
+
+    try:
+        data = QueueFreezeSchema(raw_json)
+    except voluptuous.Invalid as exc:
+        return responses.JSONResponse(status_code=400, content={"error": str(exc)})
+
+    reason = data["reason"]
+    if reason == "":
+        reason = "No freeze reason was specified."
+
+    async with github.aget_client(installation_json) as client:
+        try:
+            # Check this token as access to this repository
+            repo = typing.cast(
+                github_types.GitHubRepository,
+                await client.item(f"/repos/{owner}/{repository}"),
+            )
+        except (http.HTTPNotFound, http.HTTPForbidden, http.HTTPUnauthorized):
+            raise fastapi.HTTPException(status_code=404)
+
+        sub = await subscription.Subscription.get_subscription(
+            redis_cache, repo["owner"]["id"]
+        )
+        installation = context.Installation(
+            installation_json, sub, client, redis_cache, redis_queue
+        )
+        repository_ctxt = installation.get_repository_from_github_data(repo)
+        config_file = await repository_ctxt.get_mergify_config_file()
+
+        if config_file is not None:
+            config = get_mergify_config(config_file)
+            queue_rules = config["queue_rules"]
+
+            if all(queue_name != rule.name for rule in queue_rules):
+                return responses.JSONResponse(
+                    status_code=400,
+                    content={"error": f'The queue "{queue_name}" does not exists'},
+                )
+
+            qf = await QueueFreeze.get(repository_ctxt, queue_name)
+            if qf is not None:
+                if qf.reason != reason:
+                    qf.reason = reason
+                    await qf.save()
+                    return responses.JSONResponse(
+                        status_code=200,
+                        content={
+                            "updated_queue_freeze_data": qf.get_json(),
+                            "description": f'The queue "{queue_name}" was already frozen. Its data have been successfully updated',
```

Comment:
```suggestion
                            "description": f'The queue "{queue_name}" is already frozen. Its data have been successfully updated.',
```
Try to be consistent as if you want to use a final dot `.` or not in the description everywhere.